frontend/src/app/App.js

```diff
@@ -69,22 +68,36 @@ class App extends Component {
   componentDidMount() {
     document.addEventListener("keydown", this.escFunction, false);
     const { conferenceName } = this.props.match.params;
-    var url_string = window.location.href;
-    var url = new URL(url_string);
-    var c = url.searchParams.get("name");
+    let url_string = window.location.href;
+    let url = new URL(url_string);
+    let name = url.searchParams.get("name");
+    let showOptions = url.searchParams.get("options");
+
+    let newState = {};
+
     if (conferenceName) {
-      if (c != null) {
-        this.setState({
-          isSubmit: true,
-          form: { conferenceName: conferenceName, userName: c },
-        });
+      newState.conferenceName = conferenceName;
+      if (name !== null) {
+        newState.isSubmit = true;
+        newState.serName = name;
       } else {
-        this.setState({
-          isJoiningFromUrl: true,
-          form: { conferenceName: conferenceName },
-        });
+        newState.isJoiningFromUrl = true;
+      }
+    }
+    if (showOptions !== null) {
+      if (
+        typeof showOptions === 'string' &&
+        ['false', '0', 'no'].indexOf(showOptions.toLowerCase()) !== -1
```

Comment:
Usefull ? Cannot be converted before ?

Reply:
I guess proper solution is to use this.props.location instead window.location since we have router for this app, but I'd rather open new bug/task for that